src/Features/LanguageServer/Protocol/Handler/References/FindAllReferencesHandler.cs

```diff
@@ -0,0 +1,172 @@
+                    referenceItems.Add(referenceItem);
+                    id++;
+                };
+            }
+
+            return referenceItems.ToArrayAndFree();
+
+            static async Task<LSP.VSReferenceItem> GenerateReferenceItem(
+                int id,
+                int? definitionId,
+                DocumentSpan documentSpan,
+                SimpleFindUsagesContext context,
+                ImmutableDictionary<string, string> properties,
+                CancellationToken cancellationToken,
+                ClassifiedTextElement definitionText = null,
+                SymbolUsageInfo? symbolUsageInfo = null)
+            {
+                var location = await ProtocolConversions.DocumentSpanToLocationAsync(documentSpan, cancellationToken).ConfigureAwait(false);
+                var classifiedSpansAndHighlightSpan = await ClassifiedSpansAndHighlightSpanFactory.ClassifyAsync(documentSpan, context.CancellationToken).ConfigureAwait(false);
+                var classifiedSpans = classifiedSpansAndHighlightSpan.ClassifiedSpans;
+                var docText = await documentSpan.Document.GetTextAsync(context.CancellationToken).ConfigureAwait(false);
+
+                // TO-DO: The Origin property should be added once Rich-Nav is completed.
+                // https://github.com/dotnet/roslyn/issues/42847
+                return new LSP.VSReferenceItem
+                {
+                    ContainingMember = properties.TryGetValue(AbstractReferenceFinder.ContainingMemberInfoPropertyName, out var referenceContainingMember) ? referenceContainingMember : null,
+                    ContainingType = properties.TryGetValue(AbstractReferenceFinder.ContainingTypeInfoPropertyName, out var referenceContainingType) ? referenceContainingType : null,
+                    DefinitionId = definitionId,
+                    DefinitionText = definitionText,    // Only definitions should have a non-null DefinitionText
+                    DisplayPath = location.Uri.LocalPath,
+                    DocumentName = documentSpan.Document.Name,
+                    Id = id,
+                    Kind = symbolUsageInfo.HasValue ? ProtocolConversions.SymbolUsageInfoToReferenceKinds(symbolUsageInfo.Value) : new ReferenceKind[] { },
+                    Location = new LSP.Location { Range = location.Range, Uri = location.Uri },
+                    ProjectName = documentSpan.Document.Project.Name,
+                    ResolutionStatus = ResolutionStatusKind.ConfirmedAsReference,
+                    Text = new ClassifiedTextElement(classifiedSpans.Select(cspan => new ClassifiedTextRun(cspan.ClassificationType, docText.ToString(cspan.TextSpan)))),
+                };
+            }
+        }
+
+        private static async Task<LSP.Location[]> GetLocationsAsync(LSP.ReferenceParams request, SimpleFindUsagesContext context, CancellationToken cancellationToken)
+        {
+            var locations = ArrayBuilder<LSP.Location>.GetInstance();
```

Comment:
use using spattern.

Reply:
```suggestion
            using var _ = ArrayBuilder<LSP.Location>.GetInstance(out var locations);
```

That's the pattern i meeant :)